### ISSUE TITLE
Tipy

### ISSUE THREAD

**Issue by marhaupt** [JiMare/pizza-zadani#1]
První povinná část určitě splněna, takže úkol uznávám. U druhé je to složitější, tady jsou tipy na co dát příště pozor:
- https://github.com/JiMare/pizza-zadani/blob/b67e80917e95b6e141a6365224da0b3ba54ac925/src/components/ToppingsSelect/index.jsx#L12-L18 tady nepotřebuješ argument `checked`, stačí `newCheckedToppings[index] = !checkedToppings[index];`
- do `veganOnly` ses hrozně zamotala, protože sis asi nevšimla že v https://github.com/JiMare/pizza-zadani/blob/b67e80917e95b6e141a6365224da0b3ba54ac925/src/index.jsx#L8-L13 je u každého suroviny vlastnost `vegan`. Pak tady https://github.com/JiMare/pizza-zadani/blob/b67e80917e95b6e141a6365224da0b3ba54ac925/src/components/ToppingsSelect/index.jsx#L14 pracuješ s indexy tebou zvolených neveganských surovin (pozor, je jich tam mnohem více včetně sýrů) a jinde https://github.com/JiMare/pizza-zadani/blob/b67e80917e95b6e141a6365224da0b3ba54ac925/src/components/ToppingsSelect/index.jsx#L20-L26 se jmény, což určitě není dobře, protože to děláš na více místech a musíš si tak hlídat že ty jména/indexy použiješ vždy všechny atp. Zkus to předělat na použití vlastnosti `vegan` ze seznamu surovin, klidně to pak proletím ještě.
- chválím za použití `reduce`, dost lidí ho nechápe, já ho mám rád 👍 
- https://github.com/JiMare/pizza-zadani/blob/b67e80917e95b6e141a6365224da0b3ba54ac925/src/components/ToppingsSelect/index.jsx#L33 a ještě https://github.com/JiMare/pizza-zadani/blob/b67e80917e95b6e141a6365224da0b3ba54ac925/src/components/ToppingsSelect/index.jsx#L8 nevhodné pojmenování `e`
- https://github.com/JiMare/pizza-zadani/blob/b67e80917e95b6e141a6365224da0b3ba54ac925/src/components/ToppingsSelect/index.jsx#L8-L10 tady by věřím stačilo `toppings.fill(false)`
- https://github.com/JiMare/pizza-zadani/blob/b67e80917e95b6e141a6365224da0b3ba54ac925/src/index.jsx#L74-L76 funkci bych asi pojmenoval spíš `toggleVeganOnly`
- https://github.com/JiMare/pizza-zadani/blob/b67e80917e95b6e141a6365224da0b3ba54ac925/src/components/Check/index.jsx#L11 jsi jediná, kdo řešil druhou část a nechal tady i pro `disabled` stav třídu `check`, super 👍 

Psal jsem i připomínky na drobnosti (např. pojmenování), ostatní tím tolik "neprudím", ale vzhledem k tomu, že už v Reactu pracuješ, tak jsem chtěl dávat i takové tipy které bych třeba určitě řešil u junior developerů u nás v práci. Snad nevadí.

**Comment by JiMare** [comment]
určitě mi nevadí, že mi píšeš více věcí, spíš jsem za to ráda, děkuju, to je pravda že jsem si vůbec nevšimla, že je tam ta vlastnost vegan v objektu topping :D a říkala jsem si, že je to nějaké divné, jak to mám poznat co je vegan a co ne, tak teď je to jasné :) jediné, co asi nepůjde je použití ```toppings.fill(false)```, to jsem neznala, super dík, ale zjevně to změní pole toppings, a to není žádoucí, já potřebuji jen kopii

**Comment by marhaupt** [comment]
Jo, to už je lepší :) Ještě sem si všiml, že tady znovu vytváříš zbytečně tu samou funkci, můžeš rovnou předat dál `onChangeChecked`
https://github.com/JiMare/pizza-zadani/blob/ba61820461bf19f090c3ca2099687c2b30604baf/src/components/Topping/index.jsx#L9-L11

A s `.fill` máš pravdu, sem zapomněl že "mutuje" původní pole. Šlo by udělat `[...toppings].fill(false)` ale je to asi zbytečné.
